app/pastes/get-paste-by-hash/getPasteByHashInteractor.js

```diff
@@ -0,0 +1,42 @@
+const NotFound = require('../../errors/notFound');
+const ValidationError = require('../../errors/validationError');
+const ApiError = require('../../errors/apiError');
+
+class GetPasteByHashInteractor {
+    constructor ({presenter, validator, pasteRepository, urlRepository, responseBuilder, pasteFactory, jwt}) {
+        this.presenter = presenter;
+        this.validator = validator;
+        this.pasteRepository = pasteRepository;
+        this.urlRepository = urlRepository;
+        this.responseBuilder = responseBuilder;
+        this.pasteFactory = pasteFactory;
+        this.jwt = jwt;
+    }
+
+    async execute(request) {
+        const errors = this.validator.validate(request);
+
+        if (errors.length > 0) {
+            this.presenter.presentFailure(new ValidationError(errors));
+            return;
+        }
+        
+        const decodedToken = this.jwt.verify(request.hash, process.env.SECRET_KEY);
+
+        if (!decodedToken) {
```

Comment:
Besides payload itself, you need to be sure, that it contains the data you need (`pasteId` for example)